.github/workflows/compat.yml

```diff
@@ -10,22 +10,32 @@ on:
 
 env:
   CTEST_OUTPUT_ON_FAILURE: 1
-  PKG_CONFIG_PATH: /usr/local/opt/openssl/lib/pkgconfig
   MACOSX_DEPLOYMENT_TARGET: 10.11
 
 jobs:
   build:
     runs-on: macos-latest
     
+    env:
+        CMAKE_BUILD_DIR: ${{ github.workspace }}/build
+
     steps:
-    - uses: actions/checkout@v1
-    
+    - uses: actions/checkout@v2
+     
     - name: dependencies
       run: |
-        brew reinstall openssl@1.1
-        brew install pkgconfig doctest llvm
+        brew install llvm
         ln -s "/usr/local/opt/llvm/bin/clang-format" "/usr/local/bin/clang-format"
         ln -s "/usr/local/opt/llvm/bin/clang-tidy" "/usr/local/bin/clang-tidy"
 
+    - name: Restore cache
+      uses: actions/cache@v2
+      with:
+        path: |
+            ${{ env.CMAKE_BUILD_DIR }}/vcpkg_installed
+        key: ${{ runner.os }}-${{ hashFiles( '**/vcpkg.json' ) }}
+
     - name: build the library
-      run: make
+      run: |
+        cmake -B "${{ env.CMAKE_BUILD_DIR }}" -DCMAKE_BUILD_TYPE=Debug -DCMAKE_TOOLCHAIN_FILE="$VCPKG_INSTALLATION_ROOT/scripts/buildsystems/vcpkg.cmake" .
```

Comment:
Nit: Just to shorten this line a bit, it would be nice to push the `CMAKE_TOOLCHAIN_FILE` value to an environment variable and use the variable here.